src/test/scala/com/fulcrumgenomics/bam/PileupTest.scala

```diff
@@ -0,0 +1,178 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.fasta.ReferenceSetBuilder
+import com.fulcrumgenomics.testing.{SamRecordSetBuilder, UnitSpec}
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory
+
+class PileupTest extends UnitSpec {
+  // A reference sequence for use below
+  private val ref = {
+    val builder = new ReferenceSetBuilder()
+    builder.add("chr1").add("A", 1000)
+    builder.add("chr2").add("C", 1000)
+    val path = builder.toTempFile()
+    ReferenceSequenceFileFactory.getReferenceSequenceFile(path)
+  }
+
+  private val dict = ref.getSequenceDictionary
+
+  "PileupBuilder.build" should "build a simple pileup from fragment reads" in {
+    val piler = new PileupBuilder(dict, mappedPairsOnly=false)
+    val builder = new SamRecordSetBuilder(readLength=50, sd=Some(dict))
+    1 to 5 foreach { _ => builder.addFrag(start=100).foreach(_.setReadString("A"*50)) }
+    1 to 4 foreach { _ => builder.addFrag(start=100).foreach(_.setReadString("C"*50)) }
+    1 to 3 foreach { _ => builder.addFrag(start=100).foreach(_.setReadString("G"*50)) }
+    1 to 2 foreach { _ => builder.addFrag(start=100).foreach(_.setReadString("T"*50)) }
+
+    piler.build(builder.iterator, "chr1", 50).pile shouldBe empty
+    val pile = piler.build(builder.iterator, "chr1", 100).withoutIndels
+    pile.depth shouldBe 5 + 4 + 3 + 2
+    pile.count(_.base == 'A') shouldBe 5
+    pile.count(_.base == 'C') shouldBe 4
+    pile.count(_.base == 'G') shouldBe 3
+    pile.count(_.base == 'T') shouldBe 2
+  }
+
+  it should "build pileups from reads that contain indels" in {
+    val piler = new PileupBuilder(dict, mappedPairsOnly=false)
+    val builder = new SamRecordSetBuilder(readLength=50, sd=Some(dict))
+    builder.addFrag(name="q1", start=101, cigar="10M2D40M").foreach(_.setReadString("A"*50))
+    builder.addFrag(name="q2", start=101, cigar="10M2I38M").foreach(_.setReadString("C"*50))
+    builder.addFrag(name="q3", start=101, cigar="31M9I10M").foreach(_.setReadString("G"*50))
+    builder.addFrag(name="q4", start=101, cigar="30M9D20M").foreach(_.setReadString("T"*50))
```

Comment:
Leading insertion, since I saw some code that conditioned off of having a `0` offset insertion.